### ISSUE TITLE
False positive on `KDocReferencesNonPublicProperty`

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#8893]
## Expected Behavior
Below code should not give error for `attributes`

```
import io.opentelemetry.api.common.Attributes

/**
 * A [SpanProcessor] implementation that appends a set of [attributes][Attributes]
 * to every span. The attributes are supplied via Supplier. This Supplier may alter
 * its results and return different attributes over time. collection is mutable, and can be updated
 * by calling [.update].
 */

internal class GlobalAttributesSpanAppender {
    private val attributes: Attributes? = null
}
```

## Observed Behavior
It is giving error for `attributes`

**Comment by schalkms** [comment]
I don't get it. Why exactly should this case be ignored? Do you have any references to KDoc?

**Comment by atulgpt** [comment]
Hi @schalkms in this case, if you look into the kdoc `[attributes][Attributes]` this declaration is valid as it refers to the imported `Attributes` at `import io.opentelemetry.api.common.Attributes`

Even this declaration is correct, this rule reports a violation. 
This rule reports this as false +ve as rule only sees `[attributes]` part and matches that with the private val `private val attributes: Attributes? = null` and since that is private it is reported as violation.
But this rule doesn't look at the reference part(later one) `[Attributes]` which correctly provides the link to the mentioned variable of the kdoc
